### DISCUSSION TITLE
support dynamic schema?

### DISCUSSION THREAD

**Discussion by ductranit** [isar/isar#368] [Help] [answered]
Hi, I'm moving from Hive database. 
My project is using dynamic schema, it can be add/remove any fields when syncing from backend.
I can do it easily with sembast or hive, but I come to isar because of memory usage issue from those both databases as well as isolation.
I put a lots data into multiple tables (each table name is a key), and new table can be added at runtime as well.
My question is that isar support for dynamic schema instead of pre-defined `@Collection` schema? 
And do you think it's good for large database (around 100k - 1 million records)?

**Comment by simc** [comment, marked as answer]
Dynamic schemas are not supported unfortunately but you can transform any dynamic schema sto a static one 🙂

100k-1M records should be no problem 👍

**Reply by thipokch** [reply]
@simc Just curious like how would you transform it into a static one? Will this be supported in Isar 4, since Sqlite does actually support this.

**Reply by simc** [reply]
You can always just encode dynamic values to json and store it as String.

Isar 4 will support json natively so you can have properties of type `Map<String, dynamic>` or just `dynamic`

**Comment by bennidi** [comment]
I am also interested in the possibility of dynamic schemas. My use case is to have a content editor that allows to build different types of content (like in Drupal or the like).
Will the support of json in Isar 4 include querying / indexing of some sort?
And is a release date foreseeable?

PS Thanks a lot for building this amazing library (I love the API very much)

**Reply by simc** [reply]
Yes, there will be query support for json. No fixed release date but the first preview should be there in a few weeks. In the meantime you can store json as encoded String
